lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1039,55 +1231,470 @@ def _on_logging_relation_departed(self, event):
         self._stored.loki_push_api.pop(event.relation.id, None)
         self.on.loki_push_api_endpoint_departed.emit()
 
-    def _check_alert_rules(self, alert_groups, invalid_files) -> str:
-        """Check alert rules.
+    @property
+    def loki_push_api(self) -> List[str]:
+        """Fetch Loki Push API endpoints sent from LokiPushApiProvider through relation data.
+
+        Returns:
+            A list with Loki Push API endpoints.
+        """
+        return [
+            _type_convert_stored(loki_endpoint)
+            for loki_endpoint in self._stored.loki_push_api.values()
+        ]
+
+
+class PromtailDigestError(Exception):
+    """Raised if there is an error with Promtail binary file."""
+
+
+class LogProxyConsumer(RelationManagerBase):
```

Comment:
This should be using the consumer base class, shouldn't it? The log proxy is currently lacking support for alerts altogether 😅

Reply:
You are right, but this is a new feature. 
In this PR I only merge the 2 libs into one. I'll create an issue for this.

Reply:
It certainly will need to extend ConsumerBase. My understanding is that it will be tested and included in the next PR.